app/src/main/java/com/github/gotify/sharing/ShareActivity.java

```diff
@@ -0,0 +1,174 @@
+import androidx.appcompat.app.AppCompatActivity;
+import com.github.gotify.R;
+import com.github.gotify.Settings;
+import com.github.gotify.Utils;
+import com.github.gotify.api.Api;
+import com.github.gotify.api.ApiException;
+import com.github.gotify.api.ClientFactory;
+import com.github.gotify.client.ApiClient;
+import com.github.gotify.client.api.ApplicationApi;
+import com.github.gotify.client.api.MessageApi;
+import com.github.gotify.client.model.Application;
+import com.github.gotify.client.model.Message;
+import com.github.gotify.log.Log;
+import com.github.gotify.messages.provider.ApplicationHolder;
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.github.gotify.Utils.first;
+
+public class ShareActivity extends AppCompatActivity {
+    private ApiClient client;
+    private Settings settings;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_share);
+        Log.i("Entering " + getClass().getSimpleName());
+        setSupportActionBar(findViewById(R.id.toolbar));
+        ActionBar actionBar = getSupportActionBar();
+        if (actionBar != null) {
+            actionBar.setDisplayHomeAsUpEnabled(true);
+            actionBar.setDisplayShowCustomEnabled(true);
+        }
+        settings = new Settings(this);
+
+        // App spinner
+        client =
+                ClientFactory.clientToken(settings.url(), settings.sslSettings(), settings.token());
+        ApplicationHolder appsHolder = new ApplicationHolder(ShareActivity.this, client);
+        List<Application> apps = appsHolder.get();
+        appsHolder.onUpdate(() -> PopulateSpinner(appsHolder.get()));
+        appsHolder.request();
+
+        // Push button
+        final Button button = findViewById(R.id.button);
+        button.setOnClickListener(
+                new View.OnClickListener() {
+                    public void onClick(View v) {
+                        Message message = new Message();
+
+                        EditText edtTxtTitle = findViewById(R.id.title);
+                        String title = edtTxtTitle.getText().toString();
+                        message.setTitle(title);
+
+                        EditText edtTxtContent = findViewById(R.id.content);
+                        String content = edtTxtContent.getText().toString();
+                        message.setMessage(content);
+
+                        Spinner appSpinner = findViewById(R.id.appSpinner);
+                        String selectedApp = appSpinner.getSelectedItem().toString();
+
+                        Spinner prioritySpinner = findViewById(R.id.prioritySpinner);
+                        Long priority =
+                                Long.parseLong(prioritySpinner.getSelectedItem().toString());
+                        message.setPriority(priority);
+
+                        new GetApps(selectedApp).execute(message);
```

Comment:
in GetApps the apps will be requested via network, use appsHolder.get() instead and pass the Application Object to the async task with the message.